staging/src/k8s.io/apiextensions-apiserver/third_party/forked/celopenapi/model/schemas.go

```diff
@@ -151,11 +212,100 @@ func WithTypeAndObjectMeta(s *schema.Structural) *schema.Structural {
 	props["metadata"] = schema.Structural{
 		Generic: schema.Generic{Type: "object"},
 		Properties: map[string]schema.Structural{
-			"name": stringType,
+			"name":         stringType,
 			"generateName": stringType,
 		},
 	}
 	result.Properties = props
 
 	return result
-}
\ No newline at end of file
+}
+
+// estimateMinSizeJSON estimates the minimum size in bytes of the given schema when serialized in JSON.
+// minLength/minProperties/minItems is not currently taken into account, so the estimate is a little
+// smaller than the actual smallest size.
+func estimateMinSizeJSON(s *schema.Structural) int64 {
+	switch s.Type {
+	// take a comma into account for all cases since we'll generally call this function for arrays/maps
+	case "boolean":
+		// true,
+		return 5
+	case "number", "integer":
+		// 0,
+		return 2
+	case "string":
+		if s.ValueValidation != nil && s.ValueValidation.Format != "" && s.ValueValidation.Format != "byte" {
+			switch s.ValueValidation.Format {
+			case "duration":
+				return minDurationSizeJSON
+			case "date":
+				// add 1 to make room for a comma; since we're in estimateMinSizeJSON we need to account for
+				// this date being an array element
+				return dateSizeJSON + 1
```

Comment:
why no +1 for minDurationSizeJSON and minDatetimeSizeJSON?

Reply:
`minDurationSizeJSON` and `minDatetimeSizeJSON` already take the comma into account since they're only used in `estimateMinSizeJSON`, which is a codepath that'll only get called if we're using a duration/datetime inside an array. `dateSizeJSON` is also used from `estimateMaxSizeJSON`, whereas we have `maxDurationSizeJSON`/`maxDatetimeSizeJSON` in use for that function. I think the comments above the declarations for `minDurationSizeJSON` and `minDatetimeSizeJSON` note this, but maybe an additional comment inside `estimateMinSizeJSON` would be good?